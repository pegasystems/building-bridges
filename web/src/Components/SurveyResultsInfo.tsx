import React from 'react';
import AdminOptionsBox from './AdminOptionsBox';
interface SurveyResultsInfoProps {
    date: string;
    surveyKey: string;
    viewsNumber: number;
    questionersNumber: number;
    votersNumber: number;
    adminSecret?: string;
    surveyStateCallback: any;
    open: boolean;
}

export default class SurveyResultsInfo extends React.Component<SurveyResultsInfoProps> {

    render(): JSX.Element {
        return (
            <div className='results-info'>
                <div className='aligned-center'>
                    This page is only for <strong>{this.props.adminSecret ? 'administration of your survey': 'viewing survey results'}</strong>. If you want to ask or vote on
                    questions, go to <a
                    href={'/surveys/' + this.props.surveyKey}>this page</a>. <br/><br/>
                    Here is some information about the survey engagement: <br/>
                </div>
                <table className='results-stats-table'>
                    <thead>
                    <tr>
                        <th className='aligned-center results-stats-table'>Unique visitors</th>
                        <th className='aligned-center results-stats-table'>Unique question authors</th>
                        <th className='aligned-center results-stats-table'>Unique voters</th>
                    </tr>
                    </thead>
                    <tbody>
                    <tr>
                        <td className='aligned-center results-stats-table'>{this.props.viewsNumber}</td>
                        <td className='aligned-center results-stats-table'>{this.props.questionersNumber}</td>
                        <td className='aligned-center results-stats-table'>{this.props.votersNumber}</td>
                    </tr>
                    </tbody>
                </table>
                {this.props.adminSecret && 
                    <AdminOptionsBox open={this.props.open} 
                                     surveyStateCallback={this.props.surveyStateCallback} 
                                     surveyKey={this.props.surveyKey} 
                                     adminSecret={this.props.adminSecret}/>
                }
            </div>
        );
    }
}
