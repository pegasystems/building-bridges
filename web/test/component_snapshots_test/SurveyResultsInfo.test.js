// Link.react.test.js
import React from 'react';
import SurveyResultsInfo from '../../src/Components/SurveyResultsInfo';
import renderer from 'react-test-renderer';


test('SurveyResultsInfo renders properly test', () => {
    const component = renderer.create(
        <SurveyResultsInfo
            surveyKey='dummyKey' viewsNumber={235}
            questionersNumber={13}
            votersNumber={34}/>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});