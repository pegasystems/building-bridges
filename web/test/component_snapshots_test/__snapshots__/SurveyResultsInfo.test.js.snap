// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SurveyResultsInfo renders properly test 1`] = `
<div
  className="results-info"
>
  <div
    className="aligned-center"
  >
    This page is only for 
    <strong>
      viewing survey results
    </strong>
    . If you want to ask or vote on questions, go to 
    <a
      href="/surveys/dummyKey"
    >
      this page
    </a>
    . 
    <br />
    <br />
    Here is some information about the survey engagement: 
    <br />
  </div>
  <table
    className="results-stats-table"
  >
    <thead>
      <tr>
        <th
          className="aligned-center results-stats-table"
        >
          Unique visitors
        </th>
        <th
          className="aligned-center results-stats-table"
        >
          Unique question authors
        </th>
        <th
          className="aligned-center results-stats-table"
        >
          Unique voters
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td
          className="aligned-center results-stats-table"
        >
          235
        </td>
        <td
          className="aligned-center results-stats-table"
        >
          13
        </td>
        <td
          className="aligned-center results-stats-table"
        >
          34
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
